### PR TITLE
Update majority of `original` and `new` test beds

### PR DESCRIPTION
**Pull request by joshbruce**
**Marked version:** 0.3.19

**Markdown flavor:** Markdown.pl|CommonMark|GitHub Flavored Markdown|n/a

## Description

- Deprecates test cases from original test suite that are covered by CommonMark and GFM test suites.
- Migrates previous scenarios from the original test suite, where applicable.
- Adds verifiable "Gruber" flavor where possible (original Perl).

```js
// GFM
{
  pedantic: false,
  gfm: true
}

// CommonMark
{
  pedantic: false,
  gfm: false,
  xhtml: true,
  headerIds: false
}

// Gruber
{
  pedantic: true,
  gfm: false
}

// Issues/PRs
// For tests born from an issue or a PR, the example number will be set to the issue or PR number
```

## Need help understanding why test exists

- [ ] /new/double_link

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] Draft GitHub release notes have been updated.
- [ ] CI is green (no forced merge required).
- [ ] Merge PR

### CODE REVIEW

**Comment by joshbruce** [comment]
Well Snyk, WIP, and review guidelines seem to work. :)

**Comment by joshbruce** [comment]
This is a lot of change. I've converted and eliminated what I think is fair. Thinking we should merge this with the understanding that there will be some we might need to do differently (bringing in the full pages from Gruber) or that I need a second set of eyes on (the redos).

File: test/specs/commonmark/commonmark-spec.js [hunk 1]
```diff
@@ -147,22 +107,22 @@ describe('CommonMark 0.28 Indented code blocks', function() {
...
+    tester.test(spec, section, ignore, testOptions);
   });
 });
 
 describe('CommonMark 0.28 Fenced code blocks', function() {
   var section = 'Fenced code blocks';
 
   // var shouldPassButFails = [];
-  var shouldPassButFails = [93, 95, 96, 97, 101, 102, 106, 108, 111, 112, 113];
+  var shouldPassButFails = [88, 89, 91, 92, 93, 94, 95, 96, 97, 98, 99, 100, 101, 102, 104, 105, 106, 108, 109, 110, 111, 112, 113, 115];
```
Comment (styfle) [inline test/specs/commonmark/commonmark-spec.js]: Why are more tests failing than before?

File: test/specs/gfm/gfm-spec.js [hunk 2]
```diff
@@ -1,42 +1,27 @@
...
+  xhtml: true
 }
 
-var messenger = new Messenger();
-
 describe('GFM 0.28 Tables', function() {
   var section = 'Tables';
 
   // TODO: Verify exmaple 193 is valid and passing
+  // var shouldPassButFails = [];
```
Comment (styfle) [inline test/specs/gfm/gfm-spec.js]: You can remove this comment

File: test/specs/marked-tester.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  var markdown = 'Markdown:\n\n' + spec.markdown;
+  var html = 'Expected:\n\n' + expected;
+  var marked = 'Marked:\n\n' + actual;
+
+  return title + '\n' + ignoreString + '\n\n' + markdown + spacer + html + spacer + marked;
+}
+
+MarkedTester.test = function(spec, section, ignore, options) {
+  if (spec.section === section) {
+    var shouldFail = ~ignore.indexOf(spec.example);
```
[the remark this answers is not in the document]
Comment (joshbruce) [reply test/specs/marked-tester.js]: That's an interesting point and speaks a little to the "remove unused comments" reviews as well. 

I'm trying to consider this as a long-term tool for others. So, each test suite is a self-contained way for someone to look and know what's going on...and easily get down to work...hence the commented out empty arrays. Someone can uncomment that one, comment out the real one, and be set to go.

Definitely, considering moving the concat step to the test though. Trying to focus on 8fold over the weekend though.

File: test/specs/marked/marked-spec.js [hunk 4]
```diff
@@ -43,6 +19,118 @@ describe('Marked Code spans', function() {
...
+
+  markedSpec.forEach(function(spec) {
+    tester.test(spec, section, ignore, { gfm: false, pedantic: false });
+  });
+});
+
+describe('Marked Issues & PRs', function() {
+  var section = 'Strikethrough';
+
+  // var shouldPassButFails = [];
```
[the remark this answers is not in the document]
Comment (joshbruce) [reply test/specs/marked/marked-spec.js]: ...see last reply...Someone can uncomment that one, comment out the real one, and be set to go.

File: test/specs/marked/marked-spec.js [hunk 5]
```diff
@@ -43,6 +19,118 @@ describe('Marked Code spans', function() {
...
   markedSpec.forEach(function(spec) {
-    messenger.test(spec, section, ignore);
+    tester.test(spec, section, ignore, { gfm: false, pedantic: false });
+  });
+});
+
+describe('Marked Issues & PRs', function() {
+  var section = 'Blockquotes';
+
+  // var shouldPassButFails = [];
```
[the remark this answers is not in the document]
Comment (joshbruce) [reply test/specs/marked/marked-spec.js]: ...see last reply...Someone can uncomment that one, comment out the real one, and be set to go.

File: test/specs/gfm/gfm-spec.js (hunk 2 above)
Comment (joshbruce) [reply test/specs/gfm/gfm-spec.js]: ...see last reply...Someone can uncomment that one, comment out the real one, and be set to go.

File: test/specs/commonmark/commonmark-spec.js (hunk 1 above)
Comment (joshbruce) [reply test/specs/commonmark/commonmark-spec.js]: See `testOptions`. Before I turned `headerIds` off `xhtml` on...this time, I turned `pedantic` off, turned `gfm` off, `headerIds` off,  `xhtml` on, and changed the value for `langPrefix` to match the one from the specification.

Marked doesn't really have a "CommonMark" setting or option. Those were the combination of settings that I thought would us close enough.

Comment (joshbruce) [reply test/specs/commonmark/commonmark-spec.js]: At least that's my best guess.

**Comment by UziTech** [comment]
I'm closing this in favor of #1444

**Pull request closed without merging**